build/storage/tests/it/ptf_tests/system_tools/test_platform.py

```diff
@@ -115,46 +113,145 @@ def set_system_setup(self):
-    def create_ramdrive(self):
-        return 'Guid'
+    def create_ramdrive(self, name):
+        """Create ramdrive and attach as ns to subsystem"""
+        cmd_sender_id = (
+            self.terminal.docker.get_docker_containers_id_from_docker_image_name(
+                "cmd-sender"
+            )[0]
+        )
+        cmd = (
+            f"""docker exec {cmd_sender_id} """
+            f"""python -c 'from scripts.disk_infrastructure import create_ramdrive_and_attach_as_ns_to_subsystem; """
+            f"""print(create_ramdrive_and_attach_as_ns_to_subsystem"""
+            f"""("{self.terminal.config.ip_address}", "{name}", 4, "{NQN}", {SPDK_PORT}))'"""
+        )
+        return self.terminal.execute(cmd)
+
+    def run_storage_target_container(self, storage_dir):
+        cmd = (
+            f"cd {storage_dir} && "
+            f"AS_DAEMON=true scripts/run_storage_target_container.sh"
+        )
+        self.terminal.execute(cmd)
+        self.wait_for_running("storage-target")
 
 
 class IPUStoragePlatform(BaseTestPlatform):
-
-    def __init__(self):
-        config = IPUStorageConfig()
-        terminal = SSHTerminal(config)
+    def __init__(self, terminal):
         super().__init__(terminal)
+        # if vm is not running you have to do it
+        self.vm = VirtualMachine(terminal)
+
+    def create_virtio_blk_device(self, volume_id, physical_id):
+        """
+        :return: device handle
+        """
+        cmd_sender_id = (
+            self.terminal.docker.get_docker_containers_id_from_docker_image_name(
+                "cmd-sender"
+            )[0]
+        )
+        cmd = (
+            f"""docker exec {cmd_sender_id} """
+            f"""python -c "from scripts.disk_infrastructure import create_virtio_blk; """
+            f"""print(create_virtio_blk"""
+            f"""('{self.terminal.config.ip_address}', '{SMA_PORT}', '{self.terminal.config.ip_address}', 50051, """
+            f"""'{volume_id}', '{physical_id}', '0', '{NQN}', """
+            f"""'{self.terminal.config.ip_address}', '{NVME_PORT}'))" """
+        )
+        return self.terminal.execute(cmd)
 
-    # TODO add implementation
-    def create_virtio_blk_device(self):
-        return 'VirtioBlkDevice'
+    def delete_virtio_blk_device(self, device_handle):
+        cmd_sender_id = (
+            self.terminal.docker.get_docker_containers_id_from_docker_image_name(
+                "cmd-sender"
+            )[0]
+        )
+        cmd = (
+            f"""docker exec {cmd_sender_id} """
+            f"""python -c "from scripts.disk_infrastructure import delete_sma_device; """
+            f"""print(delete_sma_device"""
+            f"""('{self.terminal.config.ip_address}', '{SMA_PORT}', '{self.terminal.config.ip_address}', 50051, '{device_handle}'))" """
+        )
+        return self.terminal.execute(cmd)
 
+    def get_number_of_virtio_blk_devices(self):
+        return self.vm.get_number_of_virtio_blk_devices()
```

Comment:
virtual machine is not part of IPU platform. It is a part of `HostTargetPlatform` and only for KVM case.

Reply:
changed to HostTargetPlatform